src/steps/GithubActionStep.ts

```diff
@@ -0,0 +1,25 @@
+import { Step } from 'aws-cdk-lib/pipelines';
+import { JobStep } from '../workflows-model';
+
+interface GithubActionStepProps {
+  /**
+   * The Job step.
+   */
+  jobStep: JobStep;
+
+  /**
+   * Environment variables to set.
+   */
+  env?: Record<string, string>;
+}
+
+export class GithubActionStep extends Step {
+  readonly env: Record<string, string>;
+  readonly jobStep: JobStep;
+
+  constructor(id: string, { env, jobStep }: GithubActionStepProps) {
```

Comment:
in case we plan on adding more `GitHubActionStepProps` in the future, lets do

```ts
constructor(id: string, props: GitHubActionStepProps) {
  super(id);
  this.jobsStep = props.jobStep;
  this.env = props.env ?? {};
}
```

Reply:
updated